cleanlab/object_detection/summary.py

```diff
@@ -34,6 +43,412 @@
+
+    labl_freq: DefaultDict[Any, int] = collections.defaultdict(int)
+    pred_freq: DefaultDict[Any, int] = collections.defaultdict(int)
+    for sample in auxiliary_inputs:
+        _get_class_instances(sample["lab_labels"], labl_freq, class_names)
+        _get_class_instances(sample["pred_labels"], pred_freq, class_names)
+
+    labl_total, pred_total = sum(labl_freq.values()), sum(pred_freq.values())
+    return (
+        {k: round(v / labl_total, 2) for k, v in labl_freq.items()},
+        {k: round(v / pred_total, 2) for k, v in pred_freq.items()},
+    )
+
+
+def class_accuracy(
+    labels=None,
+    predictions=None,
+    *,
+    auxiliary_inputs=None,
+    class_names: Optional[Dict[Any, Any]] = None,
+    verbose=False,
+):
+    """
+    Calculate the accuracy per class.
+
+    This method can help you understand the classes on which the model performs well or poorly.
+
+    For object detection model, evaluation metrics are defined as follows:
+    - True Positive (TP): the predicted bounding box overlaps with a ground truth bounding box with the same class label
+    - False Positive (FP): the predicted bounding box overlaps with a ground truth bounding box with a different class label or does not overlap with any ground truth bounding box
+    - False Negative (FN): the ground truth bounding box does not overlap with any predicted bounding box
+    - True Negative (TN): not applicable
+
+    Parameters
+    ----------
+    labels:
+        Annotated boxes and class labels in the original dataset, which may contain some errors.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    predictions:
+        Predictions output by a trained object detection model.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    auxiliary_inputs: optional
+        Auxiliary inputs to be used in the computation of counts.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    class_names: optional
+        Optional dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``.
+
+    Returns
+    -------
+    class_metrics: Dict[Any, Dict[str, float]]
+        A dictionary mapping each class label to its metrics - TP, FP, FN, accuracy.
+    """
+    if auxiliary_inputs is None:
+        auxiliary_inputs = _get_valid_inputs_for_compute_scores(ALPHA, labels, predictions)
+    # TP: True Positive, FP: False Positive, FN: False Negative
+    class_metrics: DefaultDict[Any, Any] = collections.defaultdict(
+        lambda: {"TP": 0, "FP": 0, "FN": 0, "accuracy": 0}
+    )
+
+    for sample in auxiliary_inputs:
```

Comment:
this function is kinda complex, there are 4-5 indentations. It may be easier for debugging if it was split